src/main/resources/Diagram/WebHome.xml

```diff
@@ -66,6 +65,8 @@
 {{velocity}}
 #set ($diagramClassReference = $services.model.createDocumentReference('', 'Diagram', 'DiagramClass'))
 #if ($services.licensing.licensor.hasLicensureForEntity($diagramClassReference))
+  #includeMacros('Diagram.MigrationScriptMacros')
```

Comment:
I have some doubts about this old Velocity macro. Using the ``{{include}}`` wiki syntax macro would have been cleaner, but it requires a separate Velocity block so it complicates the code. I hope this old ``includeMacros`` doesn't have hidden issues.

Reply:
+1 to not use this deprecated way. What's the problem with:

```
{{velocity}}
#set ($diagramClassReference = $services.model.createDocumentReference('', 'Diagram', 'DiagramClass'))
#if ($services.licensing.licensor.hasLicensureForEntity($diagramClassReference))
  {{include reference='Diagram.MigrationScriptMacros'/}}
```

?

Reply:
@vmassol as I said, you need to do the include in a separate Velocity block otherwise the macro call ``#selectDiagramsRequiringMigration`` (from the next line) is evaluated before the wiki syntax (include macro) is executed and the macro definition is not found.

Reply:
ok, was just looking at the diff. There's a real question as to whether we want to keep velocity macros for include or not. Right now they're marked deprecated on their extension page, see https://extensions.xwiki.org/xwiki/bin/view/Extension/Include%20Form%20Macro

Right now we're in a grey area: we say they're deprecated but we also consider (see your comments above) that it's ok to use them because they serve some use case better than the rendering macro. 

I think we need a thread on the list to clarify this.

Reply:
I had doubts as well with this macro but since it was not marked as deprecated in 'macros.vm', I thought it might still be acceptable (I had not checked the documentation page that you mention Vincent).

I pushed a [new version](https://github.com/xwikisas/application-diagram/pull/35/commits/97b382bfea427b2ebef92de8b7c4fb419b52da45) with the following changes:
- First of all it fixes an issue with the migration script itself (in the previous commit it was not working anymore, sorry for not testing further beforehand).
- The migration macros are now always included, and executed only if a valid license is available.
- Minor formatting and wording changes were brought.

Note that the migration utility in this version works only for an upgrade of the pro version or an upgrade to it.

Please let me know if the latest commit can be merged or if you notice any issue including formatting ones.